targets/charge_controller/main.cpp

```diff
@@ -60,83 +52,217 @@ struct CANInterruptParams {
+
+extern "C" int16_t COParaDefault(CO_PARA* pg) { return 0; }
+
+extern "C" void COPdoTransmit(CO_IF_FRM* frm) {}
+
+extern "C" int16_t COPdoReceive(CO_IF_FRM* frm) { return 0; }
+
+extern "C" void COPdoSyncUpdate(CO_RPDO* pdo) {}
+
+extern "C" void COTmrLock(void) {}
+
+extern "C" void COTmrUnlock(void) {}
+
+extern "C" void HAL_CAN_RxFifo1FullCallback(CAN_HandleTypeDef* hcan) {
+    log::LOGGER.log(log::Logger::LogLevel::DEBUG, "RX Full");
+}
+
+IO::GPIO* devices[1];
+
 int main() {
     platform::init();
 
-    // Will store CANopen messages that will be populated by the EVT-core CAN
-    // interrupt
-    EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage> canOpenQueue;
-
     // Pin initialization
-    IO::UART& uart = IO::getUART<UART_TX_PIN, UART_RX_PIN>(9600);
-    IO::CAN& can = IO::getCAN<CAN_TX_PIN, CAN_RX_PIN>();
+    IO::UART& uart = IO::getUART<IO::Pin::UART_TX, IO::Pin::UART_RX>(9600);
+    uart.printf("Uart Init\n\r");
+    log::LOGGER.setUART(&uart);
 
     // GPIO pins
-    IO::GPIO& relayControl =
-        IO::getGPIO<RELAY_CTL_PIN>(IO::GPIO::Direction::OUTPUT);
     IO::GPIO& statusLED = IO::getGPIO<LED_PIN>(IO::GPIO::Direction::OUTPUT);
     IO::GPIO* bmsOK[] = {
         &IO::getGPIO<BATTERY_1_OK>(IO::GPIO::Direction::INPUT),
         &IO::getGPIO<BATTERY_2_OK>(IO::GPIO::Direction::INPUT)};
+    uart.printf("GPIO Init\n\r");
 
     // Buttons
-    Debounce standbyButton(
-        IO::getGPIO<STANDBY_BUTTON_PIN>(IO::GPIO::Direction::INPUT));
-    Debounce startButton(
-        IO::getGPIO<START_BUTTON_PIN>(IO::GPIO::Direction::INPUT));
+    IO::GPIO& standbyButtonGPIO = IO::getGPIO<STANDBY_BUTTON_PIN>(IO::GPIO::Direction::INPUT);
+    IO::GPIO& startButtonGPIO = IO::getGPIO<START_BUTTON_PIN>(IO::GPIO::Direction::INPUT);
+
+    //    DEV::Button standbyButton = DEV::Button(standbyButtonGPIO);
+    DEV::Button startButton = DEV::Button(startButtonGPIO);
+
+    uart.printf("Buttons Init\n\r");
+
     // LCD
-    IO::GPIO& LCDRegisterSEL =
-        IO::getGPIO<LCD_A0_PIN>(IO::GPIO::Direction::OUTPUT);
-    IO::GPIO& LCDReset = IO::getGPIO<LCD_RST_PIN>(IO::GPIO::Direction::OUTPUT);
-    IO::GPIO* devices[] = {
-        &IO::getGPIO<SPI_CS_PIN>(IO::GPIO::Direction::OUTPUT),
-    };
+    IO::GPIO& LCDRegisterSEL = IO::getGPIO<IO::Pin::PB_0>(IO::GPIO::Direction::OUTPUT);
+    IO::GPIO& LCDReset = IO::getGPIO<IO::Pin::PA_1>(IO::GPIO::Direction::OUTPUT);
+
+    devices[0] = &IO::getGPIO<IO::Pin::PA_0>(EVT::core::IO::GPIO::Direction::OUTPUT);
     devices[0]->writePin(IO::GPIO::State::HIGH);
 
-    IO::SPI& spi = IO::getSPI<SPI_CLK_PIN, SPI_MOSI_PIN>(devices, 1);
+    uart.printf("LCD Init\n\r");
+
+    // Spi
+    IO::SPI& spi = IO::getSPI<IO::Pin::PA_5, IO::Pin::PA_7>(devices, 1);
+    spi.configureSPI(SPI_SPEED, SPI_MODE0, SPI_MSB_FIRST);
+    uart.printf("Spi Init\n\r");
+
+    // Initialize CAN, add an IRQ which will add messages to the queue above
+    IO::CAN& can = IO::getCAN<IO::Pin::PA_12, IO::Pin::PA_11>();
```

Comment:
For readability and to follow the format of the existing code, store the pins in a constexpr with the rest.